### PR TITLE
Update Locale and LanguageIdentifier comparison names

### LINKED ISSUE #1709: Should PartialEq<str> for Locale be strict or lenient?
Background: https://github.com/unicode-org/icu4x/pull/1704

### PR DESCRIPTION
**Pull request by robertbastian**
Fixes #1709

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by zbraniecki — DISMISSED** [review]

File: components/locid/src/extensions/mod.rs [hunk 1]
```diff
@@ -150,13 +150,13 @@ impl Extensions {
     /// only_unicode
     ///     .extensions
     ///     .retain_by_type(|t| t == ExtensionType::Unicode);
-    /// assert_eq!(only_unicode, "und-u-world");
+    /// assert_eq!(only_unicode, "und-u-world".parse().unwrap());
```
Comment (sffc) [inline components/locid/src/extensions/mod.rs:153]: Suggestion (optional, here and elsewhere): I would slightly prefer

```suggestion
    /// assert_eq!(only_unicode.to_string(), "und-u-world");
```

I see you did this in some places but not others.

File: provider/datagen/src/transform/cldr/locale_canonicalizer/aliases.rs [hunk 2]
```diff
@@ -145,7 +145,9 @@ impl From<&cldr_serde::aliases::Resource> for AliasesV1<'_> {
                         (language, None, Some(region), false)
                             if language == language!("sgn")
                                 && !replacement.language.is_empty()
-                                && replacement == replacement.language.as_str() =>
+                                && replacement.script.is_none()
+                                && replacement.region.is_none()
+                                && replacement.variants == subtags::Variants::new() =>
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/locale_canonicalizer/aliases.rs:150]: Suggestion: `.variants` refers to `&[Variant]`, so you can call `.is_empty()` on it

```suggestion
                                && replacement.variants.is_empty() =>
```

File: provider/core/src/resource.rs [hunk 3]
```diff
@@ -425,9 +425,9 @@ impl From<&Locale> for ResourceOptions {
 }
 
 impl ResourceOptions {
-    pub fn cmp_bytes(&self, other: &[u8]) -> Ordering {
+    pub fn strict_cmp(&self, other: &[u8]) -> Ordering {
```
Comment (sffc) [inline provider/core/src/resource.rs:428]: Thought: I'm not sure if we want to bring the change into `ResourceOptions` as well, but I guess we probably do.

**Review by sffc — APPROVED** [review]

File: components/locid/src/extensions/mod.rs (hunk 1 above)
Comment (robertbastian) [reply components/locid/src/extensions/mod.rs:153]: I think I only did this in places where there was already a check for structural equality.

**Comment by sffc** [comment]
I'm merging this because I don't want to have a confict with it when I send the PR for #243

**Pull request merged**